src/Match2Cases.jl

```diff
@@ -143,6 +141,15 @@ function build_state_machine(location::LineNumberNode, mod::Module, value, body)
 
     state = BinderState(mod)
     entry::CodePoint = build_state_machine_core(value, source_cases, location, state)
```

Comment:
Is this purposefully a conversion, or intended more as a type assertion?

```
julia> function f(x)
          # convert x to an Int
          y::Int = x
          return y
       end
f (generic function with 2 methods)

julia> function g(x)
          # assert x is an Int
          y = x::Int
          return y
       end
g (generic function with 2 methods)

julia> f(1)
1

julia> g(1)
1

julia> f("a")
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Int64
Closest candidates are:
  convert(::Type{T}, ::T) where T<:Number at number.jl:6
  convert(::Type{T}, ::Number) where T<:Number at number.jl:7
  convert(::Type{T}, ::Base.TwicePrecision) where T<:Number at twiceprecision.jl:273
  ...
Stacktrace:
 [1] f(x::String)
   @ Main ./REPL[95]:2
 [2] top-level scope
   @ REPL[99]:1

julia> g("a")
ERROR: TypeError: in typeassert, expected Int64, got a value of type String
Stacktrace:
 [1] g(x::String)
   @ Main ./REPL[96]:2
 [2] top-level scope
   @ REPL[100]:1
```

Reply:
It is just documentation, since that is the correct type in this code.